### PR TITLE
feat: introduce input variants placeholders

### PR DESCRIPTION
**Pull request by devodii**

### CODE REVIEW

File: libs/ui/design-system/src/lib/input.tsx [hunk 1]
```diff
@@ -1,28 +1,66 @@
...
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      verified: '', // green borders
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/input.tsx:12]: **suggestion:** Those are not variants, but styles that we can target with `:user-invalid` and `:user-valid`. Same thing with `disabled`.

File: libs/ui/design-system/src/lib/input.tsx [hunk 2]
```diff
@@ -1,28 +1,66 @@
...
+      unVerified: '', // red borders
+      disabled: '',
+    },
+  },
+})
 
-interface InputProps extends React.ComponentProps<'input'> {
+type Orientation = 'horizontal' | 'vertical'
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/input.tsx:19]: **suggestion:** Orientation **is** variant, it shouldn't be passed as different prop. That's what cva is for. I bet this is where we need `mixinProps`.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/design-system/src/lib/input.tsx (hunk 2 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/input.tsx:19]: So, we need to merge that proposal to the main branch to see what effect it has :)

File: libs/ui/design-system/src/lib/input.tsx (hunk 1 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/input.tsx]: I just removed them :)

File: libs/ui/design-system/src/lib/input.tsx (hunk 2 above)
Comment (BleedingDev) [reply libs/ui/design-system/src/lib/input.tsx:15]: Will check it out and merge. :)

Comment (BleedingDev) [reply libs/ui/design-system/src/lib/input.tsx:15]: Also I am not sure... do we have horizontal / vertical in the design system? Now I don't see them, but in the past they were there, so I asked the designer.

File: libs/ui/design-system/src/lib/input.tsx [hunk 3]
```diff
@@ -1,28 +1,62 @@
...
 import { Label } from '@nmit-coursition/ui/primitives/label'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    /** todo: add orientation to the variants with mixins. */
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/input.tsx:10]: **suggestion:** Now that we don't have any variants we might use `orientation` straight away.

**suggestion:** I'd like to introduce one more called `grow` which will signalize if it should stay minimal or grow to the maximum size of its container. I have example in branch `add-design-system-button`.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/design-system/src/lib/input.tsx (hunk 2 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/input.tsx:15]: I was referring to the old primitive page, haha

File: libs/ui/design-system/src/lib/input.tsx (hunk 3 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/input.tsx]: orientations are not useful in this case

**Pull request closed without merging**